### PR TITLE
[INLONG-7339][Sort] Adjust the Sort structure for multiple versions of Apache Flink

### LINKED ISSUE #7339: [Improve][Sort] Adjust the Sort structure for multiple versions of Apache Flink
### Description

To adjust the Sort structure for multiple versions of Apache Flink and decouple Flink

### InLong Component

InLong Sort

### Are you willing to submit PR?

- [X] Yes, I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by GanfengTan**
This improve adjusts the Sort structure for multiple versions of Apache Flink.

- Fixes #7339 

### Motivation

To decouple version of Flink.

### Modifications

Move Sort Connectors

### Verifying this change

*(Please pick either of the following options)*

- [x] This change is a trivial rework/code cleanup without any test coverage.

- [ ] This change is already covered by existing tests, such as:
  *(please describe tests)*

- [ ] This change added tests and can be verified as follows:

### CODE REVIEW

**Comment by dockerzhang** [comment]
@GanfengTan please add the plan detail about how to support multiple versions of Apache Flink, thanks.

**Comment by dockerzhang** [comment]
@featzhang PTAL, thanks.

**Review by yunqingmoswu — APPROVED** [review]

File: inlong-sort/sort-flink/v1.13/pom.xml [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+    <modelVersion>4.0.0</modelVersion>
+    <parent>
+        <groupId>org.apache.inlong</groupId>
+        <artifactId>sort-flink</artifactId>
+        <version>1.6.0-SNAPSHOT</version>
+        <relativePath>../pom.xml</relativePath>
+    </parent>
+
+    <artifactId>v1.13</artifactId>
```
Comment (dockerzhang) [inline inlong-sort/sort-flink/v1.13/pom.xml:28]:
```suggestion
    <artifactId>sort-flink-v1.13</artifactId>
```

File: inlong-sort/sort-flink/v1.13/pom.xml [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+    <packaging>pom</packaging>
+
+    <name>Apache InLong - Sort-flink-v1.13</name>
```
Comment (dockerzhang) [inline inlong-sort/sort-flink/v1.13/pom.xml:31]:
```suggestion
    <name>Apache InLong - Sort Flink v1.13</name>
```

File: inlong-sort/sort-flink/pom.xml [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+    <parent>
+        <groupId>org.apache.inlong</groupId>
+        <artifactId>inlong-sort</artifactId>
+        <version>1.6.0-SNAPSHOT</version>
+        <relativePath>../pom.xml</relativePath>
+    </parent>
+
+    <artifactId>sort-flink</artifactId>
+    <packaging>pom</packaging>
+    <name>Apache InLong - Sort-flink</name>
```
Comment (dockerzhang) [inline inlong-sort/sort-flink/pom.xml:30]:
```suggestion
    <name>Apache InLong - Sort Flink</name>
```

File: inlong-sort/sort-flink/pom.xml [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+    <modules>
+        <module>v1.13</module>
```
Comment (dockerzhang) [inline inlong-sort/sort-flink/pom.xml:32]:
```suggestion
        <module>sort-flink-v1.13</module>
```

File: inlong-sort/pom.xml [hunk 5]
```diff
@@ -35,10 +35,10 @@
         <module>sort-api</module>
         <module>sort-common</module>
         <module>sort-formats</module>
-        <module>sort-connectors</module>
         <module>sort-core</module>
         <module>sort-dist</module>
         <module>sort-end-to-end-tests</module>
+        <module>sort-flink</module>
```
Comment (dockerzhang) [inline inlong-sort/pom.xml:41]: please move the `sort-flink` before `sort-dist`.

**Comment by dockerzhang** [comment]
@GanfengTan you can run `mvn spotless:apply` to fix the format error.

File: inlong-sort/sort-flink/v1.13/sort-connectors/redis/pom.xml [hunk 6]
```diff
@@ -139,7 +139,9 @@
                                     <includes>
                                         <include>org/apache/inlong/**</include>
                                         <include>META-INF/services/org.apache.flink.table.factories.Factory</include>
-                                        <include>META-INF/services/org.apache.flink.streaming.connectors.redis.common.hanlder.RedisHandler</include>
+                                        <include>
```
Comment (yunqingmoswu) [inline inlong-sort/sort-flink/v1.13/sort-connectors/redis/pom.xml:142]: Please don't update the format of it.

File: inlong-sort/sort-flink/v1.13/sort-connectors/base/src/main/resources/META-INF/services/org.apache.flink.table.factories.Factory [hunk 7]
```diff
@@ -13,5 +13,5 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-org.apache.inlong.sort.base.dirty.sink.log.LogDirtySinkFactory
+org.apache.inlong.sort.base.dirty.sink.logset.LogDirtySinkFactory
```
Comment (yunqingmoswu) [inline inlong-sort/sort-flink/v1.13/sort-connectors/base/src/main/resources/META-INF/services/org.apache.flink.table.factories.Factory:16]: Makesure it is right?

**Comment by dockerzhang** [comment]
@GanfengTan after renaming the directory of sort-connectors, I think we need to change the `sort-connectors.xml` at the same time.

File: inlong-sort/sort-flink/v1.13/pom.xml (hunk 2 above)
Comment (GanfengTan) [reply inlong-sort/sort-flink/v1.13/pom.xml:31]: done

File: inlong-sort/sort-flink/pom.xml (hunk 3 above)
Comment (GanfengTan) [reply inlong-sort/sort-flink/pom.xml:30]: done

File: inlong-sort/sort-flink/pom.xml (hunk 4 above)
Comment (GanfengTan) [reply inlong-sort/sort-flink/pom.xml:32]: done

File: inlong-sort/pom.xml (hunk 5 above)
Comment (GanfengTan) [reply inlong-sort/pom.xml:41]: done

File: inlong-sort/sort-flink/v1.13/sort-connectors/base/src/main/resources/META-INF/services/org.apache.flink.table.factories.Factory (hunk 7 above)
Comment (GanfengTan) [reply inlong-sort/sort-flink/v1.13/sort-connectors/base/src/main/resources/META-INF/services/org.apache.flink.table.factories.Factory:16]: done

File: inlong-sort/sort-flink/v1.13/sort-connectors/redis/pom.xml (hunk 6 above)
Comment (GanfengTan) [reply inlong-sort/sort-flink/v1.13/sort-connectors/redis/pom.xml:142]: done

**Comment by gong** [comment]
`sort-dist`, `sort-format` and `sort-core` need add version directory.

File: inlong-sort/sort-flink/v1.13/sort-connectors/base/src/main/resources/META-INF/services/org.apache.flink.table.factories.Factory (hunk 7 above)
Comment (GanfengTan) [reply inlong-sort/sort-flink/v1.13/sort-connectors/base/src/main/resources/META-INF/services/org.apache.flink.table.factories.Factory]: It is recommended that the names of subsequent files not conflict with the gitignore contents.

**Comment by GanfengTan** [comment]
> `sort-dist`, `sort-format` and `sort-core` need add version directory.


You are correct, this involves a global Flink version change, a dedicated PR will be created to fix. 
Will add PR information to your comment.
Thank you very much @gong

**Comment by GanfengTan** [comment]
> after renaming the directory of `sort-connectors`, I think we need to change the `sort-connectors.xml` at the same time.

done @dockerzhang

**Comment by dockerzhang** [comment]
> `sort-dist`, `sort-format` and `sort-core` need add version directory.

I think it's better to reuse more modules and functions.

**Review by dockerzhang — APPROVED** [review]

**Comment by GanfengTan** [comment]
> I think it's better to reuse more modules and functions.

+1  @dockerzhang

**Review by EMsnap — APPROVED** [review]

**Review by gong — APPROVED** [review]

**Pull request merged**